## srcbot: Source verification passed

- PR: #475325
- Attribute: `jedit`
- System: `x86_64-linux`

### Step 1: Build `.src`

- **Command**: `nix-build --no-out-link --substituters '' --expr 'with import /home/jrestivo/dev/nixpkgs { system = "x86_64-linux"; config = { allowUnfree = true; }; }; jedit.src'`
- **Status**: Exit Code: 0

<details><summary>src Build Logs</summary>

```
this derivation will be built:
  /nix/store/cvs9hbpayybjdsyr1vcr2glfh40y25vq-jEdit.drv
building '/nix/store/cvs9hbpayybjdsyr1vcr2glfh40y25vq-jEdit.drv'...
structuredAttrs is enabled
exporting https://git.code.sf.net/p/jedit/jEdit (rev refs/tags/v5.7.0) into /nix/store/4hvav4abnhhsfj7jkscq096656cndhxs-jEdit
Initialized empty Git repository in /nix/store/4hvav4abnhhsfj7jkscq096656cndhxs-jEdit/.git/
remote: Enumerating objects: 1742, done.
remote: Counting objects: 100% (1742/1742), done.
remote: Compressing objects: 100% (1613/1613), done.
remote: Total 1742 (delta 256), reused 1052 (delta 77), pack-reused 0 (from 0)
Receiving objects: 100% (1742/1742), 3.51 MiB | 5.85 MiB/s, done.
Resolving deltas: 100% (256/256), done.
From https://git.code.sf.net/p/jedit/jEdit
 * tag               v5.7.0     -> FETCH_HEAD
Switched to a new branch 'fetchgit'
refer to FETCH_HEAD as its original name refs/tags/v5.7.0
removing `.git'...
/nix/store/4hvav4abnhhsfj7jkscq096656cndhxs-jEdit

```
</details>

### Step 2: Build Package

- **Command**: `nix build --print-build-logs --no-link --rebuild --impure --expr '(import /home/jrestivo/dev/nixpkgs { system = "x86_64-linux"; config = { allowUnfree = true; }; }).jedit'`
- **Status**: Exit Code: 0

<details><summary>Package Build Logs (Last 100 lines)</summary>

```
...
jedit>     [javac]                                          ^
jedit>     [javac] /build/jEdit/test/org/gjt/sp/util/StringListTest.java:76: warning: [deprecation] join(Collection<String>,String) in StringList has been deprecated
jedit>     [javac]              assertEquals("a", StringList.join(List.of("a"), ","));
jedit>     [javac]                                          ^
jedit>     [javac] 4 warnings
jedit> 
jedit> test:
jedit>     [mkdir] Created dir: /build/jEdit/build/test/raw-reports
jedit>     [junit] Running org.gjt.sp.jedit.MiscUtilitiesTest
jedit>     [junit] Tests run: 1, Failures: 0, Errors: 0, Skipped: 1, Time elapsed: 0.133 sec
jedit>     [junit] Running org.gjt.sp.jedit.TextUtilitiesTest
jedit>     [junit] Tests run: 2, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 0.01 sec
jedit>     [junit] Running org.gjt.sp.jedit.buffer.ContentManagerTest
jedit>     [junit] Tests run: 4, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 0.016 sec
jedit>     [junit] Running org.gjt.sp.jedit.syntax.SyntaxUtilitiesTest
jedit>     [junit] Tests run: 3, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 0.004 sec
jedit>     [junit] Running org.gjt.sp.jedit.textarea.FirstLineTest
jedit>     [junit] Tests run: 3, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 1.305 sec
jedit>     [junit] Running org.gjt.sp.jedit.textarea.LineCharacterBreakerTest
jedit>     [junit] Tests run: 2, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 0.085 sec
jedit>     [junit] Running org.gjt.sp.jedit.textarea.RangeMapTest
jedit>     [junit] Tests run: 8, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 0.025 sec
jedit>     [junit] Running org.gjt.sp.jedit.textarea.ScreenLineManagerTest
jedit>     [junit] Tests run: 30, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 0.053 sec
jedit>     [junit] Running org.gjt.sp.util.GenericGUIUtilitiesTest
jedit>     [junit] Fontconfig error: Cannot load default config file: No such file: (null)
jedit>     [junit]
jedit>     [junit] Fontconfig error: No writable cache directories
jedit>     [junit]      /var/cache/fontconfig
jedit>     [junit]      /homeless-shelter/.cache/fontconfig
jedit>     [junit]
jedit>     [junit] Fontconfig error: No writable cache directories
jedit>     [junit]      /var/cache/fontconfig
jedit>     [junit]      /homeless-shelter/.cache/fontconfig
jedit>     [junit]
jedit>     [junit] Fontconfig error: No writable cache directories
jedit>     [junit]      /var/cache/fontconfig
jedit>     [junit]      /homeless-shelter/.cache/fontconfig
jedit>     [junit]
jedit>     [junit] Fontconfig error: No writable cache directories
jedit>     [junit]      /var/cache/fontconfig
jedit>     [junit]      /homeless-shelter/.cache/fontconfig
jedit>     [junit]
jedit>     [junit] Fontconfig error: No writable cache directories
jedit>     [junit]      /var/cache/fontconfig
jedit>     [junit]      /homeless-shelter/.cache/fontconfig
jedit>     [junit]
jedit>     [junit] Fontconfig error: No writable cache directories
jedit>     [junit]      /var/cache/fontconfig
jedit>     [junit]      /homeless-shelter/.cache/fontconfig
jedit>     [junit]
jedit>     [junit] Fontconfig error: No writable cache directories
jedit>     [junit]      /var/cache/fontconfig
jedit>     [junit]      /homeless-shelter/.cache/fontconfig
jedit>     [junit]
jedit>     [junit] Fontconfig error: No writable cache directories
jedit>     [junit]      /var/cache/fontconfig
jedit>     [junit]      /homeless-shelter/.cache/fontconfig
jedit>     [junit] Tests run: 2, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 0.054 sec
jedit>     [junit] Running org.gjt.sp.util.HtmlUtilitiesTest
jedit>     [junit] Tests run: 11, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 0.057 sec
jedit>     [junit] Running org.gjt.sp.util.IOUtilitiesTest
jedit>     [junit] Tests run: 2, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 0.005 sec
jedit>     [junit] Running org.gjt.sp.util.IntegerArrayTest
jedit>     [junit] Tests run: 4, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 0.003 sec
jedit>     [junit] Running org.gjt.sp.util.SegmentBufferTest
jedit>     [junit] Tests run: 3, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 0.002 sec
jedit>     [junit] Running org.gjt.sp.util.SegmentCharSequenceTest
jedit>     [junit] Tests run: 2, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 0.002 sec
jedit>     [junit] Running org.gjt.sp.util.StandardUtilitiesTest
jedit>     [junit] Tests run: 49, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 0.017 sec
jedit>     [junit] Running org.gjt.sp.util.StringListTest
jedit>     [junit] Tests run: 9, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 0.005 sec
jedit>     [junit] Running org.gjt.sp.util.SyntaxUtilitiesTest
jedit>     [junit] Tests run: 24, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 0.009 sec
jedit>     [junit] Running org.jedit.io.Native2ASCIIEncodingTest
jedit>     [junit] Tests run: 108, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 0.041 sec
jedit>     [mkdir] Created dir: /build/jEdit/build/test/merged-reports
jedit> [junitreport] Processing /build/jEdit/build/test/merged-reports/TESTS-TestSuites.xml to /tmp/null1954607296
jedit> [junitreport] Loading stylesheet ile:/nix/store/fq5l4ax04m794ani3zrpyn7170w6qfj2-ant-1.10.15/share/ant/lib/ant-junit.jar!/org/apache/tools/ant/taskdefs/optional/junit/xsl/junit-frames.xsl
jedit> [junitreport] Transform time: 622ms
jedit> [junitreport] Deleting: /tmp/null1954607296
jedit> 
jedit> build:
jedit>       [jar] Building jar: /build/jEdit/build/jedit.jar
jedit>      [copy] Copying 354 files to /build/jEdit/build
jedit>      [copy] Copying 2 files to /build/jEdit/build/jars
jedit> 
jedit> BUILD SUCCESSFUL
jedit> Total time: 28 seconds
jedit> Running phase: installPhase
jedit> Running phase: fixupPhase
jedit> stripping java archives in /nix/store/rfn8vc2cf9d700jk6n2kky9wr197q1az-jedit-5.7.0
jedit> stripping java archive /nix/store/rfn8vc2cf9d700jk6n2kky9wr197q1az-jedit-5.7.0/share/jEdit/jedit.jar
jedit> Normalized /nix/store/rfn8vc2cf9d700jk6n2kky9wr197q1az-jedit-5.7.0/share/jEdit/jedit.jar
jedit> shrinking RPATHs of ELF executables and libraries in /nix/store/rfn8vc2cf9d700jk6n2kky9wr197q1az-jedit-5.7.0
jedit> checking for references to /build/ in /nix/store/rfn8vc2cf9d700jk6n2kky9wr197q1az-jedit-5.7.0...
jedit> patching script interpreter paths in /nix/store/rfn8vc2cf9d700jk6n2kky9wr197q1az-jedit-5.7.0
jedit> /nix/store/rfn8vc2cf9d700jk6n2kky9wr197q1az-jedit-5.7.0/bin/.jedit-wrapped: interpreter directive changed from "#!/bin/sh" to "/nix/store/rlq03x4cwf8zn73hxaxnx0zn5q9kifls-bash-5.3p3/bin/sh"
jedit> stripping (with command strip and flags -S -p) in  /nix/store/rfn8vc2cf9d700jk6n2kky9wr197q1az-jedit-5.7.0/bin
```
</details>

--- a/pkgs/by-name/je/jedit/package.nix
+++ b/pkgs/by-name/je/jedit/package.nix
@@ -1,7 +1,7 @@
 {
   lib,
   stdenv,
-  fetchsvn,
+  fetchgit,
   ant,
   jdk,
   jre,
@@ -14,14 +14,11 @@ stdenv.mkDerivation (finalAttrs: {
   pname = "jedit";
   version = "5.7.0";
 
-  src =
-    let
-      versionWithDashes = lib.replaceStrings [ "." ] [ "-" ] finalAttrs.version;
-    in
-    fetchsvn {
-      url = "https://svn.code.sf.net/p/jedit/svn/jEdit/tags/jedit-${versionWithDashes}";
-      hash = "sha256-XfYK2C0QZrg4b//1eQcUNViRthBbXV+cbcYetzw2RG8=";
-    };
+  src = fetchgit {
+    url = "https://git.code.sf.net/p/jedit/jEdit";
+    tag = "v${finalAttrs.version}";
+    hash = "sha256-Q2uarFMWXTWuJ0brw1PNS/vKWUa9gOTpD6Fumn0wMoI=";
+  };
 
   ivyDeps = stdenv.mkDerivation {
     name = "${finalAttrs.pname}-${finalAttrs.version}-ivy-deps";
@@ -107,7 +104,7 @@ stdenv.mkDerivation (finalAttrs: {
   '';
 
   meta = {
-    changelog = "${finalAttrs.src.url}/doc/CHANGES.txt";
+    changelog = "https://sourceforge.net/p/jedit/jEdit/ci/v${finalAttrs.version}/tree/doc/CHANGES.txt";
     description = "Programmer's text editor written in Java";
     homepage = "https://www.jedit.org";
     license = lib.licenses.gpl2Only;

